src/components/Chart.tsx

```diff
@@ -0,0 +1,26 @@
+import React, { useEffect, useRef } from 'react';
+
+interface ChartProps {
+  id: String;
+  embed: Embed;
+}
+
+export default function Chart({ id, embed }: ChartProps) {
+  const el = useRef();
+  useEffect(() => {
+    embed.render({
+      id,
+      element: el.current,
+    });
+  }, [id]);
```

Comment:
hmm I wonder what's best practice here as nebula includes many of the react hooks we use like `useElement` and `useEffect`. For example the core object we create includes the element.

Reply:
If I understand correctly, the useElement returns the element of the container. In this file we want the element of the tab's content.

I used the `useEffect` of `react` instead of `nebula` because the nebula hook had given me a lot of unit test bugs in the past.

Reply:
Got it.

I see, I have used react to mock the implementation of nebula hooks in unit tests. But as we are using React anyway in this file we can use it for `useEffect`